### ISSUE TITLE
[Bug] Snapshot load latency degradation

### ISSUE THREAD

**Issue by sandreim** [firecracker-microvm/firecracker#2572]
We need to investigate this new intermittent failure on x86_64:

```
_________________________ test_snapshot_resume_latency _________________________
framework/report.py:244: in func_wrapper
    original_function(*args, **kwargs))
integration_tests/performance/test_snapshot_perf.py:332: in test_snapshot_resume_latency
    test_matrix.run_test(_test_snapshot_resume_latency)
framework/matrix.py:158: in run_test
    raise self._failure_aggregator
E   framework.utils.ExceptionAggregator: 
E   LoadSnapshot latency degraded.
E   assert 8 > 8.043
```

**Comment by alindima** [comment]
Adding some extra info for debugging reasons:

Seen on m5d.metal, with host kernel 4.14, Amazon Linux 2. When trying to reproduce locally, it only failed the first time, in 10 runs

**Comment by sandreim** [comment]
Failure seems to be caused by host I/O latency. A fix has been merged in #2591 .
